src/actions/searchActions.js

```diff
@@ -0,0 +1,121 @@
+import fetch from 'cross-fetch';
+
+export const toggleSearchFilter = (filter) => {
+    return {
+        type: 'TOGGLE_SEARCH_FILTER',
+        filter,
+    };
+};
+
+export const toggleSortFilter = (id) => {
+    
+    return (dispatch, getState) => {
+        const state = getState();
+        const assets = sortByFilter(id, state.filmListReducer.assets);
+
+        dispatch({
+            type: 'TOGGLE_SORT_FILTER',
+            id,
+            assets,
+        });
+    };
+};
+
+export const searchFilmsByGenre = (genre) => {
+    return {
+        type: 'SEARCH_FILMS_BY_GENRE',
+        genre,
+    };
+};
+
+export const requestAssets = (searchValue) => {
+    return { 
+        type: 'REQUEST_ASSETS',
+        searchValue,
+    };
+};
+
+export const receiveAssets = (searchValue, json) => {
+    return {
+        type: 'RECEIVE_ASSETS',
+        searchValue,
+        assets: json.data,
+    };
+};
+
+export const requestAssetsSameGenre = (searchValue) => {
+    return { 
+        type: 'REQUEST_ASSETS_SAME_GENRE',
+        searchValue,
+    };
+};
+
+export const receiveAssetsSameGenre = (searchValue, json) => {
+    return {
+        type: 'RECEIVE_ASSETS_SAME_GENRE',
+        searchValue,
+        assets: json.data,
+    };
+};
+
+export const saveInputValue = (value) => {
+    return {
+        type: 'SAVE_INPUT_VALUE',
+        value,
+    };
+};
+
+export const invalidateParams = () => {
+    return {
+        type: 'INVALIDATE_SEARCH',
+    };
+};
+
+const shouldFetchAssets = (state) => {
```

Comment:
I think this function and shouldFetchPost too similar and can be united and moved in some util file. WDYT?